src/axom/primal/geometry/Triangle.hpp

```diff
@@ -200,78 +201,106 @@ class Triangle
   {
     /* This method returns double (instead of T) and explicitly specializes
        determinant() on type double to avoid confusion of deduced template types. */
-    const VectorType A(p, m_points[0]);
-    const VectorType B(p, m_points[1]);
-    const VectorType C(p, m_points[2]);
-
     if(NDIMS < 3)
     {
       return 0.;
     }
     else
     {
+      const VectorType pA = m_points[0] - p;
+      const VectorType pB = m_points[1] - p;
+      const VectorType pC = m_points[2] - p;
+
       // clang-format off
-      return numerics::determinant< double > ( A[0], A[1], A[2],
-                                               B[0], B[1], B[2],
-                                               C[0], C[1], C[2]);
+      return numerics::determinant< double > (pA[0], pA[1], pA[2],
+                                              pB[0], pB[1], pB[2],
+                                              pC[0], pC[1], pC[2]);
       // clang-format on
     }
   }
 
 public:
   /*!
    * \brief Returns the barycentric coordinates of a point within a triangle
-   * \return The barycentric coordinates of the triangle inside a Point<T,3>
+   *
+   * \param [in] p The point at which we want to compute barycentric coordinates
+   * \param [in] skipNormalization Determines if the result should be normalized
+   * by the area of the triangle. One might want to skip this if they were
+   * only interested in the relative weights rather than their actual amounts
+   * 
    * \pre The point lies in this triangle's plane.
-   * \post The barycentric coordinates sum to 1.
-   * Adapted from Real Time Collision Detection by Christer Ericson.
+   * \post The barycentric coordinates sum to 1 when \a skipNormalization is false
+   * Otherwise, the sum of coordinates will be proportional to area of the triangle
+   *
+   * Algorithm adapted from Real Time Collision Detection by Christer Ericson.
    */
-  Point<double, 3> physToBarycentric(const PointType& p) const
+  Point<double, 3> physToBarycentric(const PointType& p,
+                                     bool skipNormalization = false) const
   {
-    using axom::numerics::determinant;
+    // Query point needs to be in triangle's plane
     SLIC_CHECK(axom::utilities::isNearlyEqual(ppedVolume(p), 0.));
 
     Point<double, 3> bary;
 
-    Vector<double, 3> u =
-      VectorType::cross_product(VectorType(m_points[0], m_points[1]),
-                                VectorType(m_points[0], m_points[2]));
-    const double x = std::abs(u[0]);
-    const double y = std::abs(u[1]);
-    const double z = std::abs(u[2]);
-
-    double ood = 1.0 / u[2];  // compute in xy plane by default
-    int c0 = 0;
-    int c1 = 1;
-
-    if(x >= y && x >= z)
-    {
-      // compute in yz plane
-      c0 = 1;
-      c1 = 2;
-      ood = 1.0 / u[0];
-    }
-    else if(y >= x && y >= z)
-    {
-      // compute in xz plane
-      c0 = 0;
-      c1 = 2;
-      ood = -1.0 / u[1];
-    }
+    auto triArea2D =
```

Comment:
why use auto here, when the return type is clearly a double?

Reply:
I think this is a lambda, so the type has to be double. I used the `-> double` syntax to mark the return type as double.